cmd/node-termination-handler.go

```diff
@@ -22,33 +22,47 @@ import (
 
 	"github.com/aws/aws-node-termination-handler/pkg/config"
 	"github.com/aws/aws-node-termination-handler/pkg/drainer"
-	"github.com/aws/aws-node-termination-handler/pkg/ec2metadata"
+	"github.com/aws/aws-node-termination-handler/pkg/drainevent"
 )
 
 func main() {
 	setupSignalHandler()
 	nthConfig := config.ParseCliArgs()
 	drainer.InitDrainer(nthConfig)
+	drainEventStore := drainevent.Store{
```

Comment:
suggestion: a more idiomatic way of doing this would be to have a `drainevent.NewStore()` function that populated a constructed `drainevent.Store` struct. In that `NewStore` function, you could ensure that, for instance, `Store.drainEvents` is `make(map[string]*DrainEvent)` so you don't have to have a conditional like this:

```go
if s.drainEvents == nil {
```

in methods like `drainEventStore.AddDrainEvent()` below.

Reply:
Cool, yep, I like it. Fixed